exchange/exchange.go

```diff
@@ -162,7 +163,27 @@ func (e *exchange) HoldAuction(ctx context.Context, bidRequest *openrtb.BidReque
 
 		if targData != nil {
 			auc.setRoundedPrices(targData.priceGranularity)
-			cacheErrs := auc.doCache(ctx, e.cache, targData, bidRequest, 60, &e.defaultTTLs, bidCategory)
+			if len(debugLog) > 0 {
+				// Build the response so far
+				debugBidResp, _ := e.buildBidResponse(ctx, liveAdapters, adapterBids, bidRequest, resolvedRequest, adapterExtra, auc, errs)
+
+				// Remove the VAST object from the request
+				for i := 0; i < len(debugBidResp.SeatBid); i++ {
+					for j := 0; j < len(debugBidResp.SeatBid[i].Bid); j++ {
+						debugBidResp.SeatBid[i].Bid[j].AdM = ""
+					}
+				}
+
+				// Remove the resolved request due to including duplicate info
+				var respExt openrtb_ext.ExtBidResponse
```

Comment:
The unmarshalling and remarshalling is expensive. Consider writing a `buildBidDebugResponse` which can omit the resolved request entirely... and it can also process the AdM removal.

Reply:
Updated to build the debug ext only. This requires no modification, so there is just one marshal call to covert to a string.

Reply:
Sweet. If that works for you, that's a really nice optimization.

Reply:
Thanks! Just confirmed that this output is sufficient for our current needs.